xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-notifiers/xwiki-platform-notifications-notifiers-default/src/main/java/org/xwiki/notifications/notifiers/internal/email/AbstractMimeMessageIterator.java

```diff
@@ -198,32 +215,59 @@ private void onPrepare(ExtendedMimeMessage message, boolean delete)
     protected abstract List<CompositeEvent> retrieveCompositeEventList(DocumentReference user)
         throws NotificationException;
 
+    private NotificationEmailGroupingStrategy getEmailGroupingStrategy()
+    {
+        NotificationEmailGroupingStrategy strategy = this.fallbackNotificationEmailGroupingStrategy;
+        String emailGroupingStrategyHint = this.notificationConfiguration.getEmailGroupingStrategyHint();
+        if (this.componentManager.hasComponent(NotificationEmailGroupingStrategy.class, emailGroupingStrategyHint)) {
+            try {
+                strategy = this.componentManager
+                        .getInstance(NotificationEmailGroupingStrategy.class, emailGroupingStrategyHint);
+            } catch (ComponentLookupException e) {
+                this.logger.error("Error while loading NotificationEmailGroupingStrategy with hint [{}]. " +
+                                "Fallback on default strategy. Root cause: [{}]",
+                        emailGroupingStrategyHint, ExceptionUtils.getRootCauseMessage(e));
+                this.logger.debug("Root cause of the error was: ", e);
```

Comment:
Not 100% sure but can't this be simplified as the following?
```suggestion
                this.logger.error("Error while loading the document [{}]", entityReference, e);
```

Reply:
In such case you'd always display the full stack. AFAIK our best practice is to not display full stack in logs, but only the cause of errors. Hence my usage of a debug log for the stack.

Reply:
My understanding of https://dev.xwiki.org/xwiki/bin/view/Community/CodeStyle/JavaCodeStyle/#HLoggingBestPractices is that if the log level is `error` the full stacktrace must be printed.

Reply:
Hmm here I should probably have used a warning actually since there's a fallback. Will fix with that.